lib/mongoid/traversable.rb

```diff
@@ -179,8 +179,8 @@ def discriminator_value
 
     included do
       class_attribute :discriminator_key, instance_accessor: false
-
       class << self
+        remove_method :discriminator_key if method_defined?(:discriminator_key)
```

Comment:
Same here -- please add a comment explaining that the preceding `class_attribute` creates both a getter and a setter, but we need to reimplement the getter (by delegating).

Also, the `if method_defined?(...)` is unnecessary here, since we know that the preceding `class_attribute` ensures the method *is* defined. 👍